go.mod

```diff
@@ -0,0 +1,3 @@
+module github.com/scylladb/go-set
+
+require github.com/fatih/set v0.2.1
```

Comment:
I do not like it. It's only needed for the benchmark.

Reply:
This is what happened when I followed the instructions in https://github.com/golang/go/wiki/Modules I don't really like it either...

Reply:
We can probably just remove it but I am not sure how it would work for a user running `go test -bench ./...` for example.

Reply:
I'm not really following vgo advancements maybe it's smart enough to figure it out (dep can do that).

Reply:
No @myitcv just some random googling and voices in the slack. No real definite idea about it.

Reply:
@dahankzter understood. To my mind there is no real reason _not_ to commit this file. It will act as a trip wire in case any of your (one!) dependences is ever detected as being inconsistent from "before".

As discussed on Slack, also adding a link here that covers the question "why is there not a separate section for test dependencies": 

https://github.com/golang/go/issues/26913#issuecomment-411976222

Reply:
That's good, thanks @myitcv I will add it when the time comes for it. Is the file format frozen yet?

Reply:
Module support in Go 1.11 is still considered experimental, so never say never, but I would highly surprised if any changes were made that were backwards incompatible.

Reply:
Just to support and expand slightly on that comment from @myitcv:  the core go team has said  the current file format understood by vgo and go 1.11 will be understood by future tooling...  for example in this sentence at the end of the official Modules proposal document: 

> Even though the details may change, future releases will be able to consume Go 1.11-compatible source trees. For example, Go 1.12 will understand how to consume the Go 1.11 go.mod file syntax, even if by then the file syntax or even the file name has changed.

Reply:
I have rerun the procedure using the latest beta (3) and committed the go.sum file as well.

Reply:
One other quick comment is that to make it clearer to your users at a glance as to why this dependency is there, you could add a manual comment in the go.mod file. Something like: `// only imported when running benchmarks` or however you would want to characterize it to help make it clear that it is not used outside of testing or benchmarks.